### PR TITLE
Add support for npm.

### LINKED ISSUE #8: Support yarn and npm

### PR DESCRIPTION
**Pull request by Kineolyan**
This partially resolves #8. PR for yarn will come soon.

### CODE REVIEW

File: docs/source/tools/npm.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
+----
+Npm
+----
+
+Documentation
+^^^^^^^^^^^^^
+
+.. autofunction:: universions.npm.get_npm_versions
```
Comment (fabiencelier) [inline docs/source/tools/npm.rst]:
```suggestion
.. autofunction:: universions.npm.get_npm_version
```

File: docs/source/tools/npm.rst [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+Python example
+^^^^^^^^^^^^^^
+
+>>> from universions.npm import get_npm_versions
```
Comment (fabiencelier) [inline docs/source/tools/npm.rst]:
```suggestion
>>> from universions.npm import get_npm_version
```

File: docs/source/tools/npm.rst [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+>>> get_npm_versions()
```
Comment (fabiencelier) [inline docs/source/tools/npm.rst]:
```suggestion
>>> get_npm_version()
```

File: test/test_npm_version.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+"""Test the npm version."""
+
+from pathlib import Path
+
+import pytest
+
+from universions import Version
+from universions.npm import get_npm_versions
```
Comment (fabiencelier) [inline test/test_npm_version.py]:
```suggestion
from universions.npm import get_npm_version
```

File: test/test_npm_version.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+
+class TestNpm(BasicMockedPopen):
+    """Test the Npm version."""
+
+    @pytest.mark.parametrize("output,version", VERSIONS)
+    def test_default_npm_version(self, output: bytes, version: Version):
+        """Test that the correct version is returned."""
+        self.popen.set_command("npm --version", stdout=output)
+        assert get_npm_versions() == version
```
Comment (fabiencelier) [inline test/test_npm_version.py]:
```suggestion
        assert get_npm_version() == version
```

File: test/test_npm_version.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+    def test_npm_version_with_path(self):
+        """Test that the correct version is returned when specifying path."""
+        path = str(Path.home() / "npm-here" / "bin" / "npm.bin")
+        self.popen.set_command(f"{path} --version", stdout=LINUX_OUT)
+        assert get_npm_versions(npm_path=path) == Version(6, 4, 1)
```
Comment (fabiencelier) [inline test/test_npm_version.py]:
```suggestion
        assert get_npm_version(npm_path=path) == Version(6, 4, 1)
```

File: universions/cli.py [hunk 7]
```diff
@@ -4,6 +4,7 @@
 
 from universions.java import get_java_version
 from universions.node import get_node_version
+from universions.npm import get_npm_versions
```
Comment (fabiencelier) [inline universions/cli.py]:
```suggestion
from universions.npm import get_npm_version
```

File: universions/cli.py [hunk 8]
```diff
@@ -24,6 +25,7 @@ def get_self_version() -> Version:
 TOOLS = {
     "java": get_java_version,
     "node": get_node_version,
+    "npm": get_npm_versions,
```
Comment (fabiencelier) [inline universions/cli.py]:
```suggestion
    "npm": get_npm_version,
```

File: universions/npm/__init__.py [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+# NPM is simple as it immediately returns the version we need
+
+from pathlib import Path
+from typing import Optional, Union
+
+from universions._exec import exec_command
+from universions.version import Version, parse_semver
+
+
+def get_npm_versions(npm_path: Optional[Union[Path, str]] = None) -> Optional[Version]:
```
Comment (fabiencelier) [inline universions/npm/__init__.py]:
```suggestion
def get_npm_version(npm_path: Optional[Union[Path, str]] = None) -> Optional[Version]:
```

**Comment by Kineolyan** [comment]
No objections to that changes. I think I used version*s* because it was the original names of several other functions.

**Pull request merged**